blockstack/blockstackd.py

```diff
@@ -3482,6 +3499,82 @@ def run_blockstackd():
            print "Failed to sign snapshot"
            sys.exit(1)
 
+    elif args.action == 'export_migration_json':
+        path = str(args.path)
+        rc = fast_sync_import(working_dir, path, public_keys=None, num_required=0, verbose=True, delete_file=False)
+        if not rc:
+           print 'fast_sync failed'
+           sys.exit(1)
+
+        # treat this as a recovery
+        setup_recovery(working_dir)
+
+        # allow consensus_hash to be provided as a string or within a file
+        consensus_hash = args.consensus_hash
+        if os.path.exists(consensus_hash):
+            with open(consensus_hash, 'r') as file:
+                consensus_hash = file.read().replace('\n', '')
+    
+        db = get_db_state(working_dir)
+        ch = db.get_current_consensus()
+        if str(ch) != consensus_hash:
+            print "Fast-sync import does not match consensus hash!"
+            sys.exit(1)
+        block = db.get_current_block()
+
+        tmpdir = tempfile.mkdtemp('blockstack-verify-chainstate-XXXXXX')
+        db_verified = False
+        try:
+            db_verified = verify_database(consensus_hash, block, working_dir, tmpdir)
```

Comment:
Just so you know, on mainnet, this method could take _days_ to run.

Reply:
Suggestions on alternatives? It seems like this is the only way to validate

Reply:
It's the only sure-fire way, unfortunately :/  I'm bringing it up so as to set expectations when we get this running -- we might not be able to do a full verification, depending on how much time we'll have between when the node creates the final v1 export and when v2 needs to boot up (or, we'll need to structure the upgrade logic so that we have enough time).  We'll probably want to test this with Xenon first, for example.

Reply:
We solved this by making the verification only run when passed a `--verify` argument. This allows miners to generate their own exports in 10-20 minutes rather than waiting days for this to complete.

Reply:
Cool!